Client/Frontend/Shields/ShieldsViewController.swift

```diff
@@ -145,25 +216,110 @@ class ShieldsViewController: UIViewController, PopoverContentComponent {
+        
+        shieldsView.simpleShieldView.reportSiteButton.addTarget(self, action: #selector(tappedReportSiteButton), for: .touchUpInside)
+        shieldsView.reportBrokenSiteView.cancelButton.addTarget(self, action: #selector(tappedCancelReportingButton), for: .touchUpInside)
+        shieldsView.reportBrokenSiteView.submitButton.addTarget(self, action: #selector(tappedSubmitReportingButton), for: .touchUpInside)
+        
+        updateShieldBlockStats()
+        
+        navigationController?.setNavigationBarHidden(true, animated: false)
+        
+        updateToggleStatus()
+        
+        if advancedControlsShowing && shieldsUpSwitch.isOn {
+            shieldsView.advancedShieldView.isHidden = false
+            shieldsView.advancedControlsBar.isShowingAdvancedControls = true
+            updatePreferredContentSize()
+        }
+        
         shieldControlMapping.forEach { shield, toggle, option in
             toggle.valueToggled = { [unowned self] on in
-                if shield == .AllOff {
-                    // Update the content size
-                    self.updateGlobalShieldState(on, animated: true)
-                }
                 // Localized / per domain toggles triggered here
                 self.updateBraveShieldState(shield: shield, on: on, option: option)
                 self.shieldsSettingsChanged?(self)
             }
         }
     }
     
+    @objc private func shieldsOverrideSwitchValueChanged() {
+        let isOn = shieldsUpSwitch.isOn
+        self.updateGlobalShieldState(isOn, animated: true)
+        self.updateBraveShieldState(shield: .AllOff, on: isOn, option: nil)
+        self.shieldsSettingsChanged?(self)
+    }
+    
+    private var advancedControlsShowing: Bool {
+        Preferences.Shields.advancedControlsVisible.value
+    }
+    
+    @objc private func tappedAdvancedControlsBar() {
+        Preferences.Shields.advancedControlsVisible.value.toggle()
+        UIView.animate(withDuration: 0.25) {
+            self.shieldsView.advancedShieldView.isHidden.toggle()
+        }
+        updatePreferredContentSize()
+    }
+    
+    @objc private func tappedAboutShieldsButton() {
+        let aboutShields = AboutShieldsViewController(tab: tab)
+        aboutShields.applyTheme(Theme.of(tab))
+        aboutShields.preferredContentSize = preferredContentSize
+        navigationController?.pushViewController(aboutShields, animated: true)
+    }
+    
+    @objc private func tappedReportSiteButton() {
+        updateContentView(to: shieldsView.reportBrokenSiteView, animated: true)
+    }
+    
+    @objc private func tappedCancelReportingButton() {
+        updateContentView(to: shieldsView.stackView, animated: true)
+    }
+    
+    @objc private func tappedSubmitReportingButton() {
+        if let url = url {
+            WebcompatReporter.reportIssue(on: url)
```

Comment:
We pass in the UA set up in AppDelegate. I'm not sure that it matters which UA is used because the reporter endpoint only uses it to categorize the OS as iOS

Reply:
sounds good, can leave it as it is